client/my-sites/email/titan-redirector/titan-redirector.jsx

```diff
@@ -0,0 +1,91 @@
+/**
+ * External dependencies
+ */
+import React, { Component } from 'react';
+import { localize } from 'i18n-calypso';
+import { connect } from 'react-redux';
+import page from 'page';
+
+/**
+ * Internal dependencies
+ */
+import wp from 'calypso/lib/wp';
+import EmptyContent from 'calypso/components/empty-content';
+import { isUserLoggedIn } from 'calypso/state/current-user/selectors';
+import { getManagePurchaseUrlFor } from 'calypso/my-sites/purchases/paths';
+
+class TitanRedirector extends Component {
+	state = {
+		loaded: false,
+		hasError: false,
+		action: null,
+		orderId: null,
+		subscriptionId: null,
+		error: null,
+	};
+
+	componentDidMount() {
+		const { mode, jwt } = this.props;
+
+		wp.undocumented()
+			.getTitanIncomingURL( mode, jwt )
+			.then(
+				( data ) => {
+					this.setState( {
+						hasError: false,
+						loaded: true,
+						action: data?.action,
+						orderId: data?.order_id,
+						subscriptionId: data?.subscription_id,
+					} );
+				},
+				( error ) => {
+					this.setState( {
+						hasError: true,
+						loaded: true,
+						error: error?.message,
+					} );
+				}
+			);
+	}
+
+	render() {
+		const { translate, isLoggedIn } = this.props;
+		const { loaded, hasError, error, action, subscriptionId } = this.state;
+
+		if ( ! isLoggedIn ) {
+			return (
+				<EmptyContent
+					title={ translate( 'You need to be logged in WordPress.com to open this page' ) }
+					action={ 'whaat' }
+				/>
+			);
+		}
+
+		if ( ! loaded ) {
+			return <EmptyContent title={ translate( 'Loading…' ) } />;
+		}
+
+		if ( loaded && hasError ) {
+			return <EmptyContent title={ translate( 'Oops - an error occurred' ) } line={ error } />;
+		}
+
+		if ( ! subscriptionId ) {
+			return <EmptyContent title={ translate( 'No subscription found' ) } />;
```

Comment:
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **44** times:
`translate( 'No Subscriptions found' )` *ES Score: 9*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/titan-incoming-urls)